trin-core/src/utils/mod.rs

```diff
@@ -4,3 +4,4 @@ pub mod db;
 pub mod node_id;
 pub mod portal_wire;
 pub mod provider;
+pub mod transport;
```

Comment:
We don't have clear architecture patterns and it is confusing where to put those kinds of types but I think his utils module is more for helper methods used for some kind of calculations/conversions.

I would put this type in `trin-core/src/cli.rs` for now. We are planning to decomposite `trin-core` in smaller crates and at some point, we would like all the CLI stuff to be in its own crate. Then having a `types.rs` file in `cli` crate seems reasonable.